test/testenv/secretutil.go

```diff
@@ -50,52 +63,52 @@ func EncodeBase64(str string) string {
 }
 
 // GetSecretObject Gets the secret object
-func GetSecretObject(deployment *Deployment, ns string, secretName string) *SecretResponse {
-	output, err := exec.Command("kubectl", "get", "secret", secretName, "-n", ns, "-o", "jsonpath='{.data}'").Output()
+func GetSecretObject(deployment *Deployment, ns string, secretName string) SecretResponse {
+	output, err := exec.Command("kubectl", "get", "secret", secretName, "-n", ns, "-o", "json").Output()
+	restResponse := SecretResponse{}
 	if err != nil {
-		cmd := fmt.Sprintf("kubectl get secret %s -n %s -o jsonpath='{.data}'", secretName, ns)
+		cmd := fmt.Sprintf("kubectl get secret %s -n %s -o json", secretName, ns)
 		logf.Log.Error(err, "Failed to execute command", "command", cmd)
-		return nil
+		return restResponse
 	}
 	// Parse response into response struct
-	restResponse := SecretResponse{}
-	err = json.Unmarshal([]byte(strings.Trim(string(output), "'")), &restResponse)
+	err = json.Unmarshal([]byte(output), &restResponse)
 	if err != nil {
 		logf.Log.Error(err, "Failed to parse response")
-		return nil
+		return restResponse
 	}
-	return &restResponse
+	return restResponse
 }
 
 // GetSecretKey Gets the value to specific key from secret object
 func GetSecretKey(deployment *Deployment, ns string, key string, secretName string) string {
 	restResponse := GetSecretObject(deployment, ns, secretName)
-	logf.Log.Info("Secret object encoded values", string(secretName))
-	//return key based on request
-	switch key {
-	case "hec_token":
-		value := DecodeBase64(restResponse.HecToken)
-		return value
-	case "idxc_secret":
-		value := DecodeBase64(restResponse.IdxcSecret)
-		return value
-	case "pass4SymmKey":
-		value := DecodeBase64(restResponse.Pass4SymmKey)
-		return value
-	case "password":
-		value := DecodeBase64(restResponse.Password)
-		return value
-	case "shc_secret":
-		value := DecodeBase64(restResponse.ShcSecret)
-		return value
-	default:
-		return "Invalid Key"
+	if restResponse == (SecretResponse{}) {
+		return "Not Found"
+	}
+	logf.Log.Info("Get secret object encoded value", "Secret Name", secretName, "Key", key)
+	value := "Invalid Key"
+	if key == "hec_token" {
```

Comment:
Nit: Use else if or switch? Can take it up later as well.

Reply:
Will do.